File: backend/backend/src/SignalingComponent/Common/Api/DTOs/RemoteTokenMessage.cs
using System.Text.Json.Serialization;
using backend.WebSocketComponent.Common.Api.DTOs;

namespace backend.SignalingComponent.Common.Api.DTOs;

public class RemoteTokenMessage : ITypedMessage
{
    public static string TypeString => "remote-token";

    [JsonIgnore]
    public string InstanceTypeString => TypeString;

    [JsonPropertyName("requestID")]
    public string? RequestId { get; set; }

    [JsonPropertyName("remoteToken")]
    public required string RemoteToken { get; set; }
}

File: frontend/src/types/token/RemoteTokenMessage.ts
import { ITypedMessage } from "../ITypedMessage";
import { MessageType } from "../MessageType";

export class RemoteTokenMessage implements ITypedMessage {
    public readonly type = MessageType.REMOTE_TOKEN;
    public msg: {
        requestID?: string;
        remoteToken: string;
    };

    public constructor(msg: { requestID?: string; remoteToken: string }) {
        this.msg = msg;
    }
}
